app/uk/gov/hmrc/eusubsidycompliancefrontend/services/EscService.scala

```diff
@@ -83,7 +83,7 @@ class EscService @Inject() (
     classTag: ClassTag[A] // is used but can appear hidden
   ): A = {
     r.fold(
-      _ => sys.error(s"Error executing $action"),
+      error => throw new RuntimeException(s"Error executing $action", error),
```

Comment:
Why throw away cause ?
"I had problem but I am not going to tell you" is not very nice